homeassistant/components/glances/__init__.py

```diff
@@ -50,8 +61,31 @@ async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     return unload_ok
 
 
-def get_api(hass: HomeAssistant, entry_data: dict[str, Any]) -> Glances:
+async def get_api(hass: HomeAssistant, entry_data: dict[str, Any]) -> Glances:
     """Return the api from glances_api."""
     entry_data.pop(CONF_NAME, None)
+    entry_data.pop(CONF_VERSION, None)
+
     httpx_client = get_async_client(hass, verify_ssl=entry_data[CONF_VERIFY_SSL])
-    return Glances(httpx_client=httpx_client, **entry_data)
+    for version in (3, 2):
+        api = Glances(httpx_client=httpx_client, version=version, **entry_data)
```

Comment:
It's safer to specifically pass each keyword instead of unpacking the config entry data. The config entry data is stored on disk and may be older or newer than the code that is using it. We have config entry versions to deal with this, but it's good practice when dealing with disk stored data to not unpack it implicitly into keywords.